src/app/views/user/user-settings/user-details/user-details.component.ts

```diff
@@ -2,33 +2,92 @@ import {Component, Input} from '@angular/core';
 import {AuthManager} from '../../../../services/auth-manager.service';
 import {User} from '../../../../models/user';
 import {UserProxy} from '../../../../services/proxy/user-proxy.service';
+import {FormGroup, FormBuilder, Validators} from '@angular/forms';
 
 @Component({
   selector: 'user-details',
-  templateUrl: './user-details.component.html'
+  templateUrl: './user-details.component.html',
+  styleUrls: ['./user-details.component.scss']
 })
 export class UserDetailsComponent {
-  editMode: boolean = false;
 
   @Input() user: User;
-  errors: any = {};
 
-  constructor(private authManager: AuthManager,
-              private userProxy: UserProxy) {
+  settingsForm: FormGroup;
+
+  saveSuccess: boolean;
+  errorMessage: string;
+  errorCause: string;
+
+  constructor(private authManager: AuthManager, private userProxy: UserProxy, private formBuilder: FormBuilder) {
+    this.settingsForm = formBuilder.group({
+      'first-name-input': [null, Validators.compose([Validators.required, Validators.minLength(2)])],
+      'last-name-input': [null, Validators.compose([Validators.required, Validators.minLength(2)])],
+      'email-input': [null, Validators.compose([Validators.required])],
+      'phone-input': [null, Validators.compose([Validators.required, Validators.minLength(9), Validators.pattern(/\+46.*/)])],
+      'new-password-input': [null, Validators.compose([Validators.minLength(6)])],
+      'old-password-input': [null],
+      'repeat-password-input': [null]
+    });
+  }
+
+  passwordsSupplied() {
+    return (this.user.newPassword || this.user.repeatedPassword) && true;
+  }
+
+  passwordsSuppliedAndMisMatch() {
+    return this.passwordsSupplied() && this.user.newPassword !== this.user.repeatedPassword && true;
   }
 
-  onEditClick() {
-    this.editMode = true;
+  handleServerErrors(errors, addPrefix: boolean = true) {
+    this.errorMessage = 'user.settings.form.submit.error';
+    if (errors.details) {
+      if (Object.keys(errors.details)[0]) {
+        if (addPrefix) {
+          this.errorCause = Object.keys(errors.details)[0].split('_').join(' ') + ' ' + errors.details[Object.keys(errors.details)[0]];
```

Comment:
@bvangelov Yes, we could do that. So the server error should be cleared on touch of the input field then?